hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -899,11 +900,26 @@ public String getMultipartKeyFSO(String volume, String bucket, String key, Strin
     final long volumeId = getVolumeId(volume);
     final long bucketId = getBucketId(volume,
             bucket);
-    long parentId =
-            OMFileRequest.getParentID(volumeId, bucketId, key, this);
-
-    String fileName = OzoneFSUtils.getFileName(key);
+    long parentId;
+    try {
+      parentId = OMFileRequest.getParentID(volumeId, bucketId, key, this);
+    } catch (final Exception e) {
+      // It is possible we miss directories and exception is thrown.
+      // see https://issues.apache.org/jira/browse/HDDS-11784
+      LOG.warn("Got exception when finding parent id for {}/{}/{}. Use another way to get it",
```

Comment:
This might be a common occurance in a concurrent system and might not warrant a `WARN` log.

Reply:
per @ivandika3 's suggestion, we probably need to update the logic of direction deletions for incomplete MPU so that we can prevent missing parent directories case to happen (or make it in a very low probability): https://github.com/apache/ozone/pull/7566#issuecomment-2584786432

From high level, I feel if this case happens frequently, means something is not right (either at code level or design level). How can a key exists and its parent directories already got deleted? Thats why I feel it is good to have this warn log here. Pls let me know your thoughts.